Assets/Scripts/Swarms/SwarmMember.cs

```diff
@@ -1,205 +1,236 @@
+        var members = controller.GetNeighboursUsingShortList(neighboursShortList, this, conf.alignmentRadius);
+        if (members.Count == 0)
+            return alignVector;
+        foreach(var member in members)
+        {
+            if (IsInFOV(rigidBody.position))
+                alignVector += (Vector3)member.rigidBody.velocity;
+        }
+        return alignVector.normalized;
+    }
+
+    Vector3 Separation(List<SwarmMember> neighboursShortList)
+    {
+        Vector3 separateVector = new Vector3();
+        var members = controller.GetNeighboursUsingShortList(neighboursShortList, this, conf.separationRadius);
+        if (members.Count == 0)
+            return separateVector;
+
+        foreach(var member in members)
+        {
+            if (IsInFOV(rigidBody.position))
+            {
+                Vector3 movingTowards = this.rigidBody.position - member.rigidBody.position;
+                if (movingTowards.magnitude > 0)
+                {
+                    separateVector += movingTowards.normalized / movingTowards.magnitude;
+                }
+            }
+        }
+        return separateVector.normalized;
+    }
+
+    //Vector3 Avoidance(List<SwarmMember> neighboursShortList)
+    //{
+    //    Vector3 avoidVector = new Vector3();
+    //    var enemyList = controller.GetEnemies(this, conf.avoidanceRadius);
+    //    if (enemyList.Count == 0)
+    //        return avoidVector;
+    //    foreach(var enemy in enemyList)
+    //    {
+    //        avoidVector += RunAway(enemy.rigidBody.position);
+    //    }
+    //    return avoidVector.normalized;
+    //}
+
+    Vector3 RunAway(Vector3 target)
+    {
+        Vector3 neededVelocity = (((Vector3)rigidBody.position) - target).normalized * conf.maxForce;
+        return neededVelocity - (Vector3)rigidBody.velocity;
+    }
+
+    virtual protected Vector3 MoveTowardsPlayer()
+    {
+        var radii = new List<float>() { conf.cohesionRadius, conf.alignmentPriority, conf.separationPriority, conf.avoidancePriority};
+        var maxRadius = radii.Max();
+        var neighboursShortList = controller.GetNeighbours(this, maxRadius);
+
+        Vector3 finalVec =
+            conf.cohesionPriority * Cohesion(neighboursShortList) +
+            conf.alignmentPriority * Alignment(neighboursShortList) +
+            conf.separationPriority * Separation(neighboursShortList);
+        //conf.avoidancePriority * Avoidance();
+        if (HasLineOfSightOnTarget())
+        {
+            //Attack
+            var targetDirection = Vector3.Normalize(target.position - transform.position);
+            finalVec += conf.pathfindPriority * targetDirection;
+        }
+        else
+        {
+            finalVec += conf.pathfindPriority* FollowSwarmLeader();
+        }
+        return finalVec;
+    }
+
+    public bool HasLineOfSightOnTarget()
+    {
+        int layerMask = LayerMask.GetMask("Abilities", "Bullets", "Floor", "Enemies");

```

Comment:
This will cause great pain..... Maybe have the layer mask created in init. Or even pass in to prefab and create in init if null.

Reply:
make private